frontend/common/src/components/tabs/Tab.tsx

```diff
@@ -0,0 +1,151 @@
+import React from "react";
+
+export type TabColor = "primary" | "secondary";
+type TabAppearance = "active" | "inactive";
+
+const styleMap: Record<
+  TabColor,
+  Record<TabAppearance, Record<string, string>>
+> = {
+  primary: {
+    active: {
+      "data-h2-font-color": "b(lightpurple)",
+      "data-h2-font-weight": "b(bold)",
+    },
+    inactive: {
+      "data-h2-font-color": "b(black)",
+    },
+  },
+  secondary: {
+    active: {
+      "data-h2-font-color": "b(lightnavy)",
+      "data-h2-font-weight": "b(bold)",
+    },
+    inactive: {
+      "data-h2-font-color": "b(black)",
+    },
+  },
+};
+export interface TabProps extends React.HTMLProps<HTMLElement> {
+  icon?: JSX.Element;
+  iconOpen?: JSX.Element;
+  iconClosed?: JSX.Element;
+  iconPosition?: "left" | "right";
+  text?: string;
+  variant?: "normal" | "close" | "label";
+  placement?: "default" | "end";
+  /* below props are injected by parent TabSet  */
+  isTabSetOpen?: boolean;
+  isTabSelected?: boolean;
+  onSelect?: VoidFunction;
+  onToggleOpen?: VoidFunction;
+  color?: TabColor;
+}
+
+export const Tab: React.FC<TabProps> = ({
+  icon,
+  iconOpen,
+  iconClosed,
+  iconPosition = "left",
+  text,
+  variant = "normal",
+  placement = "default",
+  isTabSetOpen,
+  isTabSelected,
+  onSelect,
+  onToggleOpen,
+  color = "primary",
+}): React.ReactElement => {
+  // start by calculating the icon to show
+  let effectiveIcon;
+  if (isTabSetOpen && iconOpen) {
+    effectiveIcon = iconOpen;
+  } else if (!isTabSetOpen && iconClosed) {
+    effectiveIcon = iconClosed;
+  } else {
+    effectiveIcon = icon;
+  }
```

Comment:
I tried to structure these blocks as a `const xxx = ` structure to be more functional but I couldn't quite figure it out.  Doing a `switch(true)` or an IIFE just seemed overly complicated.  I'd love to do something like [C#'s relational patterns](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#relational-patterns).

Reply:
Yeah, I also wish JS supported that pattern. What you have is clean and understandable though.